### PR TITLE
[Tests-Only]Test sharing refactor

### PR DESCRIPTION
**Pull request by jasson99**
## Description:
This PR refactors the sharing test file to use provider verify in the tests.

## Related Issue:
#643

### CODE REVIEW

File: tests/sharingTest.js [hunk 1]
```diff
@@ -68,77 +79,65 @@ describe('Main: Currently testing file/folder sharing,', function () {
...
+        provider.removeInteractions().then(done, done.fail)
+      })
       // needs to be double checked ...
       describe('updating share permissions,', function () {
-        beforeAll(async function (done) {
-          const promises = []
-          promises.push(setGeneralInteractions(provider))
-          promises.push(provider.addInteraction(
+        it('confirms not changed permissions', async function (done) {
+          oc = await createOwncloud()
```
Comment (dpakach) [inline tests/sharingTest.js:138]: create oc Instance in beforeAll hook

**Review by dpakach — COMMENTED** [review]
everything looks fine, please squash your commits

File: tests/sharingTest.js [hunk 2]
```diff
@@ -606,9 +368,10 @@ describe('Main: Currently testing file/folder sharing,', function () {
         }
       })
 
-      it('checking method : getShares for shared file', function (done) {
+      it('checking method : getShares for shared file', async function (done) {
```
Comment (individual-it) [inline tests/sharingTest.js:371]: no await in this function as far as I see

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by dpakach** [comment]
@swoichha all looks ok, please fix the commit name

**Comment by swoichha** [comment]
> @swoichha all looks ok, please fix the commit name

Done

**Review by individual-it — APPROVED** [review]

**Pull request merged**